src/Akka.Streams.Kafka.Tests/BugFix240SupervisionStrategy.cs

```diff
@@ -0,0 +1,434 @@
+                        .WithAttributes(ActorAttributes.CreateSupervisionStrategy(Decider)), 
+                    Materializer);
+
+            await GuardWithTimeoutAsync(sourceTask, TimeSpan.FromSeconds(5));
+            
+            var settings = CreateConsumerSettings<Null, string>(group1).WithValueDeserializer(new StringDeserializer());
+            var probe = KafkaConsumer
+                .PlainSource(settings, Subscriptions.Assignment(new TopicPartition(topic1, 0)))
+                .Select(c => c.Value)
+                .RunWith(this.SinkProbe<string>(), Materializer);
+
+            probe.Request(10);
+            for (var i = 0; i < 9; i++)
+            {
+                var message = probe.ExpectNext();
+                Log.Info($"> [{i}]: {message}");
+            }
+            callCount.Should().Be(1);
+            probe.Cancel();
+        }
+        
+        [Fact]
+        public async Task SupervisionStrategy_Decider_on_PlainSource_should_work()
+        {
+            var callCount = 0;
+            Directive Decider(Exception cause)
+            {
+                callCount++;
+                if(cause is ConsumeException ex && ex.Error.IsSerializationError())
+                {
+                    return Directive.Resume;
+                }
+                return Directive.Stop;
+            }
+
+            int elementsCount = 10;
+            var topic1 = CreateTopic(1);
+            var group1 = CreateGroup(1);
+
+            var sourceTask = ProduceStrings(new TopicPartition(topic1, 0), Enumerable.Range(1, elementsCount), ProducerSettings);
+
+            await GuardWithTimeoutAsync(sourceTask, TimeSpan.FromSeconds(3));
+            
+            var settings = CreateConsumerSettings<int>(group1).WithValueDeserializer(Deserializers.Int32);
+
+            var probe = KafkaConsumer
+                .PlainSource(settings, Subscriptions.Assignment(new TopicPartition(topic1, 0)))
+                .WithAttributes(ActorAttributes.CreateSupervisionStrategy(Decider))
+                .Select(c => c.Value)
+                .RunWith(this.SinkProbe<int>(), Materializer);
+
+            probe.Request(elementsCount);
+            probe.ExpectNoMsg(TimeSpan.FromSeconds(10));
+            callCount.Should().Be(elementsCount);
+            probe.Cancel();
+        }        
+        
+        [Fact]
+        public async Task SupervisionStrategy_Decider_on_PlainSource_should_stop_on_internal_error()
+        {
+            int elementsCount = 10;
+            var topic1 = CreateTopic(1);
+            var group1 = CreateGroup(1);
+
+            var sourceTask = ProduceStrings(new TopicPartition(topic1, 0), Enumerable.Range(1, elementsCount), ProducerSettings);
+
+            await GuardWithTimeoutAsync(sourceTask, TimeSpan.FromSeconds(3));
+            
+            var settings = CreateConsumerSettings<Null, string>(group1).WithAutoCreateTopicsEnabled(false);
+
+            var probe = KafkaConsumer
```

Comment:
Need to explain that we're subscribing to a non-existent partition, which creates the resulting error from the Confluent driver

Reply:
Done